demo-app/src/androidTest/java/com/worldpay/access/checkout/sample/card/standard/testutil/NewCardFragmentTestUtils.kt

```diff
@@ -0,0 +1,197 @@
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isChecked
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.isEnabled
+import androidx.test.espresso.matcher.ViewMatchers.isNotChecked
+import androidx.test.espresso.matcher.ViewMatchers.isNotEnabled
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withParent
+import androidx.test.espresso.matcher.ViewMatchers.withTagValue
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import com.worldpay.access.checkout.sample.MainActivity
+import com.worldpay.access.checkout.sample.R
+import com.worldpay.access.checkout.sample.testutil.NewAbstractFragmentTestUtils
+import com.worldpay.access.checkout.sample.testutil.UITestUtils.uiObjectWithId
+import com.worldpay.access.checkout.ui.AccessCheckoutEditText
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers
+
+class NewCardFragmentTestUtils(activityRule: ActivityScenarioRule<MainActivity>) : NewAbstractFragmentTestUtils(activityRule) {
+
+    enum class Input {
+        PAN, CVC, EXPIRY_DATE
+    }
+
+    fun isInInitialState() = apply {
+        progressBarNotVisible()
+        enabledStateIs(pan = true, cvc = true, expiryDate = true, submitButton = false)
+        cardDetailsAre(pan = "", cvc = "", expiryDate = "")
+        hasNoBrand()
+        paymentsCvcSessionCheckedState(checked = false)
+    }
+
+    fun setPaymentsCvcSwitchState(checked: Boolean) = apply {
+        wait2 { onView(withId(R.id.card_flow_payments_cvc_switch)).check(matches(isDisplayed())) }
+
+        if (checked != withId(R.id.card_flow_payments_cvc_switch).matches(isChecked()))
+            uiObjectWithId(R.id.card_flow_payments_cvc_switch).click()
+    }
+
+    fun paymentsCvcSessionCheckedState(checked: Boolean) = apply {
+        val checkedViewMatcher = if (checked) isChecked() else isNotChecked()
+        wait2 { onView(withId(R.id.card_flow_payments_cvc_switch)).check(matches(isDisplayed())) }
+        wait2 { onView(withId(R.id.card_flow_payments_cvc_switch)).check(matches(checkedViewMatcher)) }
+    }
+
+    fun requestIsInProgress() = apply {
+        progressBarIsVisible()
+        enabledStateIs(pan = false, cvc = false, expiryDate = false, paymentsCvcSwitch = false, submitButton = false)
+    }
+
+    fun hasResponseDialogWithMessage(response: String) = apply {
+        dialogHasText(response)
+    }
+
+    fun hasErrorDialogWithMessage(error: String) = apply {
+        dialogHasText(error)
+    }
+
+    fun closeDialog() = apply {
+        onView(withId(android.R.id.button1)).perform(click())
+    }
+
+    fun isInErrorState(pan: String? = null, cvc: String? = null, expiryDate: String? = null) = apply {
```

Comment:
I don't see anything in the body of the function that checks that something is in error state? Are we missing something or does the function need renaming?

Reply:
This is inherited from the class that already exists in `master` to avoid any regression